### ISSUE TITLE
`get_correlated_variable_rows` in chapter 14.3 has dead code

### ISSUE THREAD

**Issue by krassowski** [jokergoo/ComplexHeatmap-reference#10]
In https://jokergoo.github.io/ComplexHeatmap-reference/book/more-examples.html#visualize-cell-heterogeneity-from-single-cell-rnaseq in the function `get_correlated_variable_genes`:

```R
get_correlated_variable_genes = function(mat, n = nrow(mat), cor_cutoff = 0, n_cutoff = 0) {
    ind = order(apply(mat, 1, function(x) {
            q = quantile(x, c(0.1, 0.9))
            x = x[x < q[1] & x > q[2]]
            var(x)/mean(x)
        }), decreasing = TRUE)[1:n]
    mat2 = mat[ind, , drop = FALSE]
    dt = cor(t(mat2), method = "spearman")
    diag(dt) = 0
    dt[abs(dt) < cor_cutoff] = 0
    dt[dt < 0] = -1
    dt[dt > 0] = 1

    i = colSums(abs(dt)) > n_cutoff

    mat3 = mat2[i, ,drop = FALSE]
    return(mat3)
}
```

The fragment that filters by quantile does not seem to do anything. `q[1]` is the lower quantile, `q[2]` is the upper quantile of `x`, thus `x < q[1] & x > q[2]]` is always `FALSE`.  This means that `x` gets length of 0, and the entire expression:

```R
apply(mat, 1, function(x) {
  q = quantile(x, c(0.1, 0.9))
  x = x[x < q[1] & x > q[2]]
  var(x)/mean(x)
})
```

returns a named vector with "NA" only. In consequence `ind` is equivalent to `1:n`.

What was meant probably was `x = x[x > q[1] & x < q[2]]`, but if the example works without this step, it could just be removed.

If we correct the direction to align with the likely intent:

| Before (note: 721 genes) | After correcting directions (note: 693 genes) |
| -- | -- |
| ![image](https://user-images.githubusercontent.com/5832902/179030867-46de4326-511b-4c0d-bcd2-8df375a75ef9.png) | ![image](https://user-images.githubusercontent.com/5832902/179030716-bb7a08cc-fb19-4f3d-801a-049f28259aed.png) |

After removing the filtering code:

```
get_correlated_variable_genes = function(mat, n = nrow(mat), cor_cutoff = 0, n_cutoff = 0) {
    dt = cor(t(mat), method = "spearman")
    diag(dt) = 0
    dt[abs(dt) < cor_cutoff] = 0
    dt[dt < 0] = -1
    dt[dt > 0] = 1

    i = colSums(abs(dt)) > n_cutoff

    mat3 = mat[i, ,drop = FALSE]
    return(mat3)
}
```

![image](https://user-images.githubusercontent.com/5832902/179031474-34baf254-7c3b-4d99-930f-4bf325d474a2.png)

**Comment by jokergoo** [comment]
You are right. That was a mistake. It is interesting that `order()` on a vector of all `NA` values still returns a meaningful ordering.

```r
> order(c(NA, NA, NA))
[1] 1 2 3
```

I will adjust the code.

Thanks!
